### ISSUE TITLE
Revise protocol documentation

### ISSUE THREAD

**Issue by Askaholic** [FAForever/server#482]
We really need a dedicated server protocol document where we write down what messages are sent by the client, and what responses come from the server. That way the client developers can have something to reference without needing to read the server code. This would also push us toward defined protocol versions where we can explicitly say "version 1.0 has exactly these commands".

Ideally we would refactor the server so that much of this documentation could be generated automatically from doc strings.

**Comment by KaukaHan** [comment]
how about implementing the hole protocol into a webassembly module, witch can be shared between the server and client ?

**Comment by KaukaHan** [comment]
https://webassembly.org/
https://wasmer.io/
tldr: webassembly and wasmer would allow to compile into a platform indepenent, efficient, sandboxed, binary format witch can be embed into other software. the biggest problem i think is, you need a compiled language to compile into wasm, like c/c++ or rust. might be overkill therefor.
